blueprints/networking/filtering-proxy-psc/main.tf

```diff
@@ -0,0 +1,300 @@
+ */
+
+###############################################################################
+#                        Host project and VPC resources                       #
+###############################################################################
+
+module "project" {
+  source          = "../../../modules/project"
+  project_create  = var.project_create != null
+  billing_account = try(var.project_create.billing_account, null)
+  parent          = try(var.project_create.parent, null)
+  name            = var.project_id
+  services = [
+    "dns.googleapis.com",
+    "compute.googleapis.com",
+    "logging.googleapis.com"
+  ]
+}
+
+module "vpc" {
+  source     = "../../../modules/net-vpc"
+  project_id = module.project.project_id
+  name       = "${var.prefix}-vpc"
+  subnets = [
+    {
+      name          = "proxy"
+      ip_cidr_range = var.cidrs.proxy
+      region        = var.region
+    }
+  ]
+  subnets_psc = [
+    {
+      name          = "psc"
+      ip_cidr_range = var.cidrs.psc
+      region        = var.region
+    }
+  ]
+}
+
+module "firewall" {
+  source     = "../../../modules/net-vpc-firewall"
+  project_id = module.project.project_id
+  network    = module.vpc.name
+  ingress_rules = {
+    allow-ingress-squid = {
+      description = "Allow squid ingress traffic"
+      source_ranges = [
+        var.cidrs.psc, "35.191.0.0/16", "130.211.0.0/22"
+      ]
+      targets              = [module.service-account-squid.email]
+      use_service_accounts = true
+      rules = [{
+        protocol = "tcp"
+        ports    = [3128]
+      }]
+    }
+  }
+}
+
+module "nat" {
+  source                = "../../../modules/net-cloudnat"
+  project_id            = module.project.project_id
+  region                = var.region
+  name                  = "default"
+  router_network        = module.vpc.name
+  config_source_subnets = "LIST_OF_SUBNETWORKS"
+  # 64512/11 = 5864 . 11 is the number of usable IPs in the proxy subnet
+  config_min_ports_per_vm = 5864
+  subnetworks = [
+    {
+      self_link            = module.vpc.subnet_self_links["${var.region}/proxy"]
+      config_source_ranges = ["ALL_IP_RANGES"]
+      secondary_ranges     = null
+    }
+  ]
+  logging_filter = var.nat_logging
+}
+
+###############################################################################
+#                              PSC resources                                  #
+###############################################################################
+
+resource "google_compute_service_attachment" "service_attachment" {
+  name                  = "psc"
+  project               = module.project.project_id
+  region                = var.region
+  enable_proxy_protocol = false
+  connection_preference = "ACCEPT_MANUAL"
+  nat_subnets           = [module.vpc.subnets_psc["${var.region}/psc"].self_link]
+  target_service        = module.squid-ilb.forwarding_rule_self_link
+  consumer_accept_lists {
+    project_id_or_num = module.project.project_id
```

Comment:
this should be project IDs of consumer projects, not the producer, isn't it? most probably coming from the variable?

Reply:
After @ludoo remark, i moved everything to own project with multiple VPCs to showcase this.